signature/policy_eval_signedby.go

```diff
@@ -70,6 +71,13 @@ func (pr *prSignedBy) isSignatureAuthorAccepted(image types.UnparsedImage, sig [
 			return PolicyRequirementError(fmt.Sprintf("Signature by key %s is not accepted", keyIdentity))
 		},
 		validateSignedDockerReference: func(ref string) error {
+			r := image.Reference().DockerReference()
+			if _, isCanonical := r.(reference.Canonical); isCanonical || reference.IsNameOnly(r) {
+				if !pr.UntaggedReference.matchesDockerReference(image, ref) {
```

Comment:
Perhaps we don’t need to support `image@digest` signatures at all, in the first iteration. So, `reject` would simply refuse all digested references.

when `acceptableSignedIdentity`, the reference must pass through `signedIdentity`; if that is an `exactReference`, it would be including the tag.